Fix passing of lease settings in typed cluster sharding.

horvath-martin opened the pull request:
References #29662

horvath-martin committed:
Fix passing of lease settings in typed cluster sharding. (#29662)

johanandren commented on akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/ClusterShardingSettings.scala:
```diff
@@ -353,7 +356,8 @@ final class ClusterShardingSettings(
       tuningParameters: ClusterShardingSettings.TuningParameters,
-      coordinatorSingletonSettings: ClusterSingletonManagerSettings) =
+      coordinatorSingletonSettings: ClusterSingletonManagerSettings,
+      leaseSettings: Option[LeaseUsageSettings]) =
```
I think we will have to add another constructor rather, to keep source/binary compatiblity, and deprecate it (that will cover the current constructor, which user code may be calling)

horvath-martin replied:
I have compiled it, but the formatter changed 66 other files.
Should I also add these files as well?
![format](https://user-images.githubusercontent.com/53307459/94434236-bef0ac80-0199-11eb-8eca-fa90abc9e038.png)

johanandren replied:
Hmmm, that is surprising, it should only change the files you have touched (and have failed the PR validation if anyone tried to commit incorrectly formatted sources just like how it did with this PR).

Skip those and only push the files you have changed, after they were reformatted.

johanandren reviewed (commented):
One thing around binary compatibility. Make sure to compile before commit so you get the code autoformatted.

horvath-martin committed:
Fix binary compatibility. (#29662)

johanandren reviewed (approved):
LGTM, thanks @horvath-martin !

patriknw commented on akka-cluster-sharding-typed/src/main/scala/akka/cluster/sharding/typed/ClusterShardingSettings.scala:
```diff
@@ -339,8 +341,37 @@ final class ClusterShardingSettings(
+    val coordinatorSingletonSettings: ClusterSingletonManagerSettings,
+    val leaseSettings: Option[LeaseUsageSettings]) {
 
+  @deprecated("Use constructor with leaseSettings", "2.6.9")
```
should be "2.6.10"

horvath-martin committed:
Fix deprecation version. (#29662)

horvath-martin commented:
Is there any further problem? The CI failed, but it seems to be a networking error. Can I somehow trigger a repeated build or is it even necessary? (The only change since the previous successful build was a text value change.)
